File: src/main/scala/resonant/core/content/debug/TileInfiniteEnergy.java
package resonant.core.content.debug;

import net.minecraftforge.common.ForgeDirection;
import resonant.lib.prefab.tile.TileElectrical;
import universalelectricity.api.energy.EnergyStorageHandler;

/** Creative mode way to quickly debug machines that use or produce electrical energy
 * 
 * @author DarkGuardsman */
public class TileInfiniteEnergy extends TileElectrical
{
    public TileInfiniteEnergy()
    {
        setEnergyHandler(new EnergyStorageHandler(Long.MAX_VALUE));
        getEnergyHandler().setMaxExtract(Long.MAX_VALUE);
        ioMap = 728;
    }

    @Override
    public void updateEntity()
    {
        super.updateEntity();
        getEnergyHandler().setEnergy(Long.MAX_VALUE);
        produce();
    }

    @Override
    public long onReceiveEnergy(ForgeDirection from, long receive, boolean doReceive)
    {
        return receive;
    }

    @Override
    public long onExtractEnergy(ForgeDirection from, long request, boolean doExtract)
    {
        return request;
    }
}

File: src/main/scala/resonant/core/content/debug/TileInfiniteEnergy.scala
package resonant.core.content.debug

import universalelectricity.api.core.grid.electric.EnergyStorageHandler
import net.minecraft.block.material.Material
import resonant.lib.content.prefab.java.TileElectricStorage

/**
 * @since 31/05/14
 * @author tgame14
 */
class TileInfiniteEnergy(mat: Material) extends TileElectricStorage(mat)
{
	energy = new EnergyStorageHandler(Double.MaxValue)
	energy.setMaxExtract(Double.MaxValue)
	ioMap = 728

	override def updateEntity()
	{
		super.updateEntity()
		energy.setEnergy(Double.MaxValue)
	}

}
